extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -0,0 +1,344 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {CSS} from '../../../build/amp-lightbox-viewer-0.1.css';
+import {isExperimentOn} from '../../../src/experiments';
+import {Layout} from '../../../src/layout';
+import {lightboxManagerForDoc} from '../../../src/lightbox-manager';
+import {user, dev} from '../../../src/log';
+import {ancestorElements} from '../../../src/dom';
+
+/** @const */
+const EXPERIMENT = 'amp-lightbox-viewer';
+
+class AmpLightboxViewer extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.NODISPLAY;
+  }
+
+  /** @override */
+  renderOutsideViewport() {
+    return true;
+  }
+
+  /** @override */
+  buildCallback() {
+    user().assert(isExperimentOn(this.win, EXPERIMENT),
+        `Experiment ${EXPERIMENT} disabled`);
```

Comment:
We usually add the element that the error is referencing `this.element` to the log as well.